pilot/pkg/xds/internalgen.go

```diff
@@ -153,3 +162,52 @@ func (sg *InternalGen) Generate(proxy *model.Proxy, push *model.PushContext, w *
 	}
 	return res
 }
+
+func (sg *InternalGen) registerWorkload(con *Connection) {
```

Comment:
we haven't had any discussion on the auto registration let alone the workload deployment thing nor do we even have a full end to end cohesive experience for VMs drafted up. Given that, I am reluctant to even have an auto registration mechanism lest it rot like the istioctl register command that was created two years ago to automatically create services, deployments! Can we please have ONE doc that outlines an authoritative VM deployment experience so that we know what we are implementing?

Reply:
I am not sure why you say we haven't discussed this. The idea has been discussed for nearly half a year, in multiple working groups and have design docs: https://docs.google.com/document/d/10TyPy-6bOs0_7JljGpc2RneDdsaR9GktjwwaaVbz6fs, https://docs.google.com/document/d/1G4s_PrmvkwCKoH_kXVeSUAszsaj3ZfpIlraSdeTSPKU/.

I would agree we don't have a single e2e doc describing the full experience

Reply:
I'm pretty sure auto-registration was discussed in the original 'mesh expansion' document, and was demoed and discussed in environments.

Reply:
BTW - I agree with Shriram on rot, if we do this we should put istioctl register out of its misery. 

Istioctl register was created because we didn't have time to properly implement auto-registration - and at the time we were storing VMs in the Services.Endpoints which was difficult to handle with kubectl due to the need to patch ( and retry ). With the introduction of ServiceEntry (and merging) it has become obsolete - and WorkloadEntry is making it completely useless.